flow/core/kernel/scenario/base.py

```diff
@@ -251,22 +253,97 @@ def gen_even_start_pos(self, initial_config, num_vehicles, **kwargs):
+        if any(lanes[0] != lanes[i] for i in range(1, len(lanes))):
+            flag = True
+
+        x = x0
+        car_count = 0
+        startpositions, startlanes = [], []
+
+        # generate uniform starting positions
+        while car_count < num_vehicles:
+            # collect the position and lane number of each new vehicle
+            pos = self.get_edge(x)
+
+            # ensures that vehicles are not placed in an internal junction
+            while pos[0] in dict(self.internal_edgestarts).keys():
+                # find the location of the internal edge in total_edgestarts,
+                # which has the edges ordered by position
+                edges = [tup[0] for tup in self.total_edgestarts]
+                indx_edge = next(
+                    i for i, edge in enumerate(edges) if edge == pos[0])
+
+                # take the next edge in the list, and place the car at the
+                # beginning of this edge
+                if indx_edge == len(edges) - 1:
+                    next_edge_pos = self.total_edgestarts[0]
+                else:
+                    next_edge_pos = self.total_edgestarts[indx_edge + 1]
+
+                x = next_edge_pos[1]
+                pos = (next_edge_pos[0], 0)
+
+            # ensures that you are in an acceptable edge
+            while pos[0] not in available_edges:
+                x = (x + self.edge_length(pos[0])) % self.length()
+                pos = self.get_edge(x)
+
+            # ensure that in variable lane settings vehicles always start a
+            # vehicle's length away from the start of the edge. This, however,
+            # prevents the spacing to be completely uniform.
+            if flag and pos[1] < VEHICLE_LENGTH:
+                pos0, pos1 = pos
+                pos = (pos0, VEHICLE_LENGTH)
+                x += VEHICLE_LENGTH
+                increment -= (VEHICLE_LENGTH * self.num_lanes(pos0)) / \
+                             (num_vehicles - car_count)
+
+            # place vehicles side-by-side in all available lanes on this edge
+            for lane in range(min([self.num_lanes(pos[0]), lanes_distr])):
+                car_count += 1
+                startpositions.append(pos)
+                startlanes.append(lane)
+
+                if car_count == num_vehicles:
+                    break
+
+            x = (x + increment + VEHICLE_LENGTH + min_gap) % self.length()
+
+        # add a perturbation to each vehicle, while not letting the vehicle
+        # leave its current edge
+        if initial_config.perturbation > 0:
+            for i in range(num_vehicles):
+                perturb = np.random.normal(0, initial_config.perturbation)
+                edge, pos = startpositions[i]
+                pos = max(0, min(self.edge_length(edge), pos + perturb))
+                startpositions[i] = (edge, pos)
+
+        return startpositions, startlanes
+
+    def gen_random_start_pos(self, initial_config, num_vehicles):
```

Comment:
Am I correct in seeing that this method does not check/resolve overlapping vehicle positions?

Reply:
Note, if not, that such a resolution can be done in N^2 where N is the number of vehicles. Doesn't necessarily have to be done now, but if not it should be added to the issues.

Reply:
creating an issue about this. This might be causing problems for some people too.